tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2267,8 +2267,8 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
           genLongZero()
         case LongLiteral(value) =>
           val (lo, hi) = LongImpl.extractParts(value)
-          genNewLong(LongImpl.initFromParts,
-              js.IntLiteral(lo), js.IntLiteral(hi))
+          js.New(encodeClassVar(LongImpl.RuntimeLongClass),
+              List(js.IntLiteral(lo), js.IntLiteral(hi)))
```

Comment:
Can we find a way that this does not blow up spectacularly if somebody has the great idea to synthesize a class in IR that inherits from `RuntimeLong`? (strictly speaking, this is legal, the IRChecker allows it).

Reply:
Hum ... add the notion of `final` in the IR? The other alternative would be to special-case `RuntimeLong` in the IR checker.

We have the same problem with `java.lang.Class`, btw.

Reply:
Probably that'll be necessary :S Not in this PR in any case :)